crates/egui/src/widgets/plot/mod.rs

```diff
@@ -1038,14 +980,20 @@ impl Plot {
     }
 }
 
+enum BoundsModification {
+    Auto,
+    Set(PlotBounds),
+    Translate(Vec2),
+}
+
 /// Provides methods to interact with a plot while building it. It is the single argument of the closure
 /// provided to [`Plot::show`]. See [`Plot`] for an example of how to use it.
 pub struct PlotUi {
     items: Vec<Box<dyn PlotItem>>,
     next_auto_color_idx: usize,
     last_screen_transform: ScreenTransform,
-    bounds_modified: AxisBools,
     response: Response,
+    bounds_modifications: Vec<BoundsModification>,
```

Comment:
why are you accumulating modifications instead of applying them right away?

Reply:
So that they can be applied together at the right time, there are some other transforms that should be done between lines 711 and 763 before the modifications.

I'm sure there are other ways to do this as well, but the bound handling is pretty complex and I didn't want to mess with it too much.

Reply:
Added a comment in https://github.com/emilk/egui/pull/2410/commits/eaec082d8ddf56450e63a12916e70be8c6cd72b2.